cms/models.py

```diff
@@ -252,6 +259,16 @@ def clean(self):
             if errors:
                 raise ValidationError(errors)
 
+    def get_context(self, request, *args, **kwargs):
+        course = CoursePage.objects.get(course=self.course)
+        course_url = course.get_url() if course else ""
```

Comment:
I think these two would never align, Or let's say if there is no course from `CoursePage.objects.get(course=self.course)` it would break with an exception before it reaches `if course`.